variables.tf

```diff
@@ -197,6 +197,11 @@ variable "private_dns_name" {
   default     = ""
 }
 
+variable "dns_zone_project_id" {
+  type        = string
+  default     = ""
+  description = "DNS zone project id"
```

Comment:
I think it's the best place to elaborate and note that this var takes precedence over network_project_id, which takes precedene over project_id